README.md

```diff
@@ -1,16 +1,115 @@
-# generator-model-sdk
+<h5 align="center">
+  <br>
+  <a href="https://www.asyncapi.org"><img src="https://github.com/asyncapi/parser-nodejs/raw/master/assets/logo.png" alt="AsyncAPI logo" width="200"></a>
+  <br>
+  AsyncAPI Model SDK
+</h5>
+<p align="center">
+  <em>The official SDK for generating data models from JSON Schema and AsyncAPI spec</em>
+</p>
 
+AsyncAPI Model SDK is a set of classes/functions for generating data models from JSON Schema and AsyncAPI spec.
 
+---
 
-## Simplification process
+## :loudspeaker: ATTENTION:
 
-In order to simplify the model rendering process as much as possible we want to simplify any input schemas into CommonModel's where combination, conditional and what ever we have are applied so we have a bare minimum schema.
+This package is under development and it has not reached version 1.0.0 yet, what means its API might change without prior notice. Once it reaches its first stable version, we'll follow semantic versioning.
 
-We have split out the different simplification processes into separate small functions which recursively simplifies schemas, these are the different simplifications:
+---
 
-- [Simplification of types](./docs/SimplifyTypes.md)
-- [Simplification of enums](./docs/SimplifyEnums.md)
-- [Simplification of items](./docs/SimplifyItems.md)
-- [Simplification of properties](./docs/SimplifyProperties.md)
-- [Simplification of additional properties](./docs/SimplifyAdditionalProperties.md)
-- [Simplification of required](./docs/SimplifyRequired.md)
+<!-- toc is generated with GitHub Actions do not remove toc markers -->
+
+<!-- toc -->
+
+- [Installation](#installation)
+- [How it works](#how-it-works)
+  * [The transformation process](#the-transformation-process)
+  * [The simplification process](#the-simplification-process)
+  * [The generatiion process](#the-generatiion-process)
+- [Example](#example)
+- [Development](#development)
+- [Contributing](#contributing)
+
+<!-- tocstop -->
+
+## Installation
+
+Run this command to install the SDK in your project:
+
+```bash
+npm install --save @asyncapi/generator-model-sdk
+```
+
+## How it works
+
+The process of creating data models from input data consists of three main process: transformation input data to JSON Schema, simplification and generatiion. 
+
+### The transformation process
+
+The transformation process starts from checking what type of input data is. Currently the AsyncAPI Model SDK supports JSON Schema and AsyncAPI spec. Each input data must be converted to the JSON Schema or be a subset of JSON Schema (like AsyncAPI spec) to go to the next process - the simplification process.
+
+### The simplification process
+
+In order to simplify the model generation process as much as possible, AsyncAPI Model SDK simplifies transformed JSON schema into CommonModels, which contain information what type of model is, what properties single model has, etc, so finally we have a bare minimum schema.
```

Comment:
```suggestion
In order to simplify the model generation process as much as possible, AsyncAPI Model SDK simplifies the transformed JSON schema into `CommonModel`s, which contain information about the type of model it is, what properties it might have, etc, so finally we have a bare minimum schema. Think of it like converting data validation rules (JSON Schema) to a data definition.
```